cmd/der2ascii/writer.go

```diff
@@ -355,6 +355,44 @@ func derToASCIIImpl(out *bytes.Buffer, in []byte, indent int, stopAtEOC bool) []
 					// Emit the remaining as a DER element.
 					derToASCIIImpl(out, elem.body[1:], indent+1, false) // Adds a trailing newline.
 					addLine(out, indent, "}")
+				} else if len(elem.body) == 1 && elem.body[0] == 0 {
+					addLine(out, indent, fmt.Sprintf("%s b`` }", header))
+				} else if len(elem.body) > 1 && len(elem.body) <= 5 && elem.body[0] < 8 {
+					// Convert to a b`` literal when the leading bit is valid and the
+					// number of data octets is at most 4.
+					//
+					// The 4 octet limit is as above: X.509 incorrectly encodes things
+					// like signatures and RSA moduli as BIT STRING, and because
+					// overly-long bit strings don't tend to be particularly helpful for
+					// readability.
+					bits := new(strings.Builder)
+
+					// The first octet is the number of unused bits.
+					significant := 8 - elem.body[0]
+					for i, octet := range elem.body[1:] {
+						// Last octet gets some special handling.
+						isLast := i == len(elem.body)-2
+						for j := 0; j < 8; j++ {
+							if isLast {
+								if octet == 0 && significant <= 0 {
+									break
+								}
+								if significant == 0 {
+									bits.WriteRune('|')
+								}
+								significant--
```

Comment:
Nit: Rather than counting down `significant`, maybe just compare `j` against `significant`? It's a hair shorter, and it feels mildly weird to set up a counter way at the begging and only touch it at `isLast`.